lib/waistband/model.rb

```diff
@@ -0,0 +1,195 @@
+require 'digest/md5'
+require 'active_support/core_ext/object/blank'
+require 'active_model'
+
+module Waistband
+  class Model
+
+    extend ::ActiveModel::Callbacks
+
+    include ::ActiveModel::Validations
+    include ::ActiveModel::Validations::Callbacks
+
+
+    #################
+    # Class methods #
+    #################
+
+    class << self
+
+      attr_reader :type_name, :column_defaults
+
+      def index_name(name)
+        @index_name = name.to_s
+      end
+
+      def index_type(name)
+        @type_name = name.to_s
+      end
+
+      def index
```

Comment:
I like that.
On Feb 22, 2015 11:31 AM, "Evan Tahler" <notifications@github.com> wrote:

> In lib/waistband/model.rb
> <https://github.com/taskrabbit/waistband/pull/18#discussion_r25135503>:
>
> > +    # Class methods #
> > +    #################
> > +
> > +    class << self
> > +
> > +      attr_reader :type_name, :column_defaults
> > +
> > +      def index_name(name)
> > +        @index_name = name.to_s
> > +      end
> > +
> > +      def index_type(name)
> > +        @type_name = name.to_s
> > +      end
> > +
> > +      def index
>
> It could be a hierarchy though! Config YMLS overwrite model mappings.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/taskrabbit/waistband/pull/18/files#r25135503>.
>